vscode-extension/src/features/treeView/assignmentExplorer/nodes/AssignmentExplorerRootNode.ts

```diff
@@ -9,67 +9,116 @@ import { CollisionsGroupNode } from "./CollisionsGroupNode";
+        } else {
+            this._label = app.name || app.manifest.name;
+            this._description = app.manifest.version;
+            this._tooltip = `${app.manifest.name} v${app.manifest.version}`;
+            this._uriAuthority = app.hash;
+        }
         this._contextValues.push(ContextValues.Sync);
 
-        this._hasLogical = this._app.config.idRanges.length > 0;
-        this._hasObject = this._app.config.objectTypesSpecified.length > 0;
+        this._hasLogical = this._apps.flatMap(app => app.config.idRanges).length > 0;
+        this._hasObject = this._apps.flatMap(app => app.config.objectTypesSpecified).length > 0;
 
         this._contextValues.push(ContextValues.Sync);
         if (!this._hasLogical && !this._hasObject) {
             this._contextValues.push(ContextValues.CopyRanges);
         }
 
-        this._subscription = app.assignmentMonitor.onAssignmentChanged(({ assigned, unassigned }) => {
-            this._assigned = assigned;
-            this._unassigned = unassigned;
-            this._view.update(this);
-        });
+        this.attachPoolApp(app);
+        this.populateAssignments(app.assignmentMonitor.lost, app.assignmentMonitor.unassigned, app.assignmentMonitor.lostFieldIds, app.assignmentMonitor.unassignedFields, app.hash);
+    }
 
-        this._assigned = app.assignmentMonitor.assigned;
-        this._unassigned = app.assignmentMonitor.unassigned;
+    private populateAssignments(lost: AssignedALObject[], unassigned: ALObject[], lostFieldIds: AssignedALObject[], unassignedFieldIds: ALObjectNamespace[], hash: string) {
+        this._lostPerApp[hash] = lost;
+        this._unassignedPerApp[hash] = unassigned;
+        this._lostFieldsPerApp[hash] = lostFieldIds;
+        this._unassignedFieldsPerApp[hash] = unassignedFieldIds;
+        this.mergeAssignments();
+    }
+
+    private mergeAssignments() {
+        this._lost = [];
+        this._unassigned = [];
+        //TODO: this._assigned not working for app pool as it should only show assigned objects of current app to see which one are lost
+        for (let hash in this._lostPerApp) {
+            this._lost.push(...this._lostPerApp[hash]);
+        }
+        for (let hash in this._unassignedPerApp) {
+            this._unassigned.push(...this._unassignedPerApp[hash]);
+        }
+
+        this._unassignedFields = [];
+        for (let hash in this._unassignedFieldsPerApp) {
+            this._unassignedFields.push(...this._unassignedFieldsPerApp[hash]);
+        }
+        this._lostFields = [];
+        for (let hash in this._lostFieldsPerApp) {
+            this._lostFields.push(...this._lostFieldsPerApp[hash]);
+        }
     }
 
     protected override getChildren(): Node[] {
         let children: Node[] = [];
 
         children.push(new CollisionsGroupNode(this, this._unassigned));
-        children.push(new LostGroupNode(this, this._assigned));
+        if (!this._apps.some(app => app.config.appPoolId))
```

Comment:
Curly braces. I maybe am not noticing them everywhere, but when I do - I put this comment 😉 Please fix these.

Reply:
Luckily there's a linter for it ;)
Done in 7ba87ccc7cdb1b45172d51296ded88b2e5128580